desc/backend.py

```diff
@@ -66,19 +66,23 @@
 )
 
 if use_jax:  # noqa: C901 - FIXME: simplify this, define globally and then assign?
-    jit = jax.jit
-    fori_loop = jax.lax.fori_loop
-    cond = jax.lax.cond
-    switch = jax.lax.switch
-    while_loop = jax.lax.while_loop
-    vmap = jax.vmap
-    bincount = jnp.bincount
-    repeat = jnp.repeat
-    take = jnp.take
-    scan = jax.lax.scan
-    from jax import custom_jvp
+    from jax import custom_jvp, jit, vmap
+
+    imap = jax.lax.map
     from jax.experimental.ode import odeint
-    from jax.scipy.linalg import block_diag, cho_factor, cho_solve, qr, solve_triangular
+    from jax.lax import cond, fori_loop, scan, switch, while_loop
+    from jax.nn import softmax
+    from jax.numpy import bincount, flatnonzero, repeat, take
+    from jax.numpy.fft import irfft, rfft, rfft2
+    from jax.scipy.fft import dct, idct
+    from jax.scipy.linalg import (
+        block_diag,
+        cho_factor,
+        cho_solve,
+        eigh_tridiagonal,
```

Comment:
i added test for reverse diff. this isn't a bottleneck. I'd rather keep it since it's mathematically more efficient, and we'll gain the benefits once JAX makes their implementation better.

Reply:
This would be huge for ballooning as well because we only need the max eigenvalues and we are currently calculating all of them. I don't think there are any plant to do this in near-future.